__tests__/api.test.js

```diff
@@ -91,4 +91,50 @@ describe("API endpoints", () => {
                 expect(articles).toBeSortedBy(expected, {descending: true})
         });
     })
+    test("/api/articles/:article_id/comments endpoint should return all article comments in an array", () => {
+        return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+            .then((result) => {
+                const comments = result.body.comments;
+                const expected = 11
+                expect(comments.length).toBe(expected)
+                expect(comments[0]).toHaveProperty('comment_id')
+                expect(comments[0]).toHaveProperty('votes')
+                expect(comments[0]).toHaveProperty('created_at')
+                expect(comments[0]).toHaveProperty('author')
+                expect(comments[0]).toHaveProperty('body')
+                expect(comments[0]).toHaveProperty('article_id')
+        });
+    })
+    test("/api/articles/:article_id/comments should return array of comments in date descending order", () => {
+        return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+            .then((result) => {
+                const comments = result.body.comments;
+                const expected = 'created_at'
+                expect(comments).toBeSortedBy(expected, {descending: true})
+        });
+    })
+    test("When article id doesn't exist for comments it should return an error", () => {
+        return request(app)
+        .get("/api/articles/100/comments")
+        .expect(400)
+            .then((result) => {
+                const message = JSON.parse(result.error.text).msg;
+                const expected = 'There are no comments for this article'
+                expect(message).toEqual(expected)
+        });
+    })
+    test("When article id for comments is not a number it should return syntax error", () => {
+        return request(app)
+        .get("/api/articles/hundred/comments")
+        .expect(400)
+            .then((result) => {
+                const message = JSON.parse(result.error.text).msg;
+                const expected = 'Invalid input syntax'
+                expect(message).toEqual(expected)
+        });
+    })
```

Comment:
Another test to consider is what if a valid article has no comments? What should this test look like?